sklearn/preprocessing/_discretization.py

```diff
@@ -184,6 +188,13 @@ def fit(self, X, y=None):
             Ignored. This parameter exists only for compatibility with
             :class:`~sklearn.pipeline.Pipeline`.
 
+        sample_weight : ndarray of shape (n_bins_,) or (n_features, n_bins_)
+            Weights (values between 0 and 1) that represent the proportion of
```

Comment:
I don't think that sample weight needs to be in [0, 1].

Reply:
I corrrected this in the following commit : https://github.com/scikit-learn/scikit-learn/pull/22048/commits/85fe2b74ee5fe571b92f113f9dd8182339a4c4aa

`sample_weight : ndarray of shape (n_samples,)
            
Contains weight values to be associated with each sample.`